src/match.c

```diff
@@ -0,0 +1,128 @@
+#include <stdio.h>
+#include <stdlib.h>
+#include <stdbool.h>
+#include <assert.h>
+#include <gmodule.h>
```

Comment:
I very strongly suggested at the start of the first sprint that y'all use the `simclist` library for lists. Why are we using `GList`?

Reply:
I decided to use GList because I'm not entirely sure how using outside libraries work, and glib is already installed in csil computers, so there's no compatibility issues. That being said, if simclist can work without compatibility issues, I'm open to changing it!

Reply:
(resolved)